lib/kvdb/kvdb_home.c

```diff
@@ -113,9 +113,26 @@ kvdb_home_pidfile_path_get(const char *home, char *buf, const size_t buf_sz)
     INVARIANT(buf);
     INVARIANT(buf_sz > 0);
 
-    int n;
+    const char *dir, *name;
+    char namebuf[32];
+    int n, rc;
+
+    rc = access(home, W_OK);
+    if (rc == -1) {
+        dir = getenv("XDG_RUNTIME_DIR");
+        if (!dir || *dir != '/')
+            dir = "/tmp";
+
+        n = snprintf(namebuf, sizeof(namebuf), "hse-%d.pid", getpid());
+        assert(n > 0 && n < sizeof(namebuf));
+
+        name = namebuf;
+    } else {
+        dir = home;
```

Comment:
Might be nice to write a comment about this fallback behavior

Reply:
Will do.